Cython/Utility/CConvert.pyx

```diff
@@ -130,3 +130,69 @@ cdef inline tuple {{to_tuple_cname}}({{base_type}} *v, Py_ssize_t length):
         Py_INCREF(value)
         PyTuple_SET_ITEM(t, i, value)
     return t
+
+#################### c_array_slice.to_py ####################
+
+cdef extern from *:
+    void Py_INCREF(object o)
+    void Py_DECREF(object o)
+    object PyErr_Format(exc, const char *format, ...)
+    int __Pyx_is_valid_index(Py_ssize_t i, Py_ssize_t limit)
+
+    list PyList_New(Py_ssize_t size)
+    void PyList_SET_ITEM(object p, Py_ssize_t pos, object o)
+
+    tuple PyTuple_New(Py_ssize_t size)
+    void PyTuple_SET_ITEM(object p, Py_ssize_t pos, object o)
+
+    set PySet_New(void *)
+    void PySet_Add(object p, object o)
```

Comment:
Change this to `int PySet_Add(object p, object o) except -1` and Cython should handle the exception case.